### PR TITLE
[stdlib] Move two more Iterators into nested types

### PR DESCRIPTION
**Pull request by airspeedswift**
This was all done a while back in 4.0/4.1 once it was possible, but these got missed.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please test source compatibility

File: stdlib/public/core/Algorithm.swift [hunk 1]
```diff
@@ -121,39 +151,13 @@ extension EnumeratedIterator: IteratorProtocol, Sequence {
...
   /// Returns an iterator over the elements of this sequence.
   @inlinable
-  public func makeIterator() -> EnumeratedIterator<Base.Iterator> {
-    return EnumeratedIterator(_base: _base.makeIterator())
+  public func makeIterator() -> Iterator {
+    return Iterator(_base: _base.makeIterator())
   }
 }
 
+@available(*, deprecated: 4.2, renamed: "EmptyCollection.Iterator")
```
Comment (jrose-apple) [inline stdlib/public/core/Algorithm.swift]: Wrong rename?

**Comment by slavapestov** [comment]
Neat!

File: stdlib/public/core/Algorithm.swift (hunk 1 above)
Comment (airspeedswift) [reply stdlib/public/core/Algorithm.swift]: 🤦🏻‍♂️thanks

**Comment by airspeedswift** [comment]
Source compatibility passed but results didn't post for some reason, I guess cos I pushed the typo fix.

**Comment by airspeedswift** [comment]
@swift-ci please smoke test and merge

**Comment by airspeedswift** [comment]
@swift-ci please smoke test

**Comment by shahmishal** [comment]
@swift-ci please smoke test and merge

**Pull request merged**